common/setup-phase/src/lib.rs

```diff
@@ -4,42 +4,29 @@ multiversx_sc::imports!();
 
 #[multiversx_sc::module]
 pub trait SetupPhaseModule {
-    #[endpoint(endSetupPhase)]
-    fn end_setup_phase(&self) {
-        self.require_caller_initiator();
-        self.require_setup_phase();
-
-        self.setup_phase_complete().set(true);
+    fn require_setup_complete(&self, caller_shard_id: u32) {
+        require!(
+            self.is_setup_phase_complete(),
+            "The setup is not completed in shard {}",
+            caller_shard_id
+        );
     }
 
     fn require_caller_initiator(&self) {
```

Comment:
Is this function needed?

Reply:
i made it an `inline` function, might as well use it to make the endpoint as short as possible